plugins/outputs/cloudwatch/convert_otel.go

```diff
@@ -206,6 +206,9 @@ func fetchEntityFields(resourceAttributes pcommon.Map) cloudwatch.Entity {
 
 // processEntityAttributes fetches the fields with entity prefix and creates an entity to be sent at the PutMetricData call.
 func processEntityAttributes(entityMap map[string]string, targetMap map[string]*string, mutableResourceAttributes pcommon.Map) {
+	entityattributes.AttributeEntityToShortNameMapRWMutex.RLock()
+	defer entityattributes.AttributeEntityToShortNameMapRWMutex.RUnlock()
```

Comment:
We use this function to process two maps. We shouldn't be locking this if we pass in the `KeyAttributeEntityShortNameMap`. We should only be calling the read lock when the `AttributeEntityShortNameMap` is passed in.

Reply:
Wonder if it makes sense to lock the function in the caller then